### PR TITLE
Snippets support angle bracket invocation

### LINKED ISSUE #420: Snippet examples don't get registered when used with angle brackets
[`ember-code-snippet`'s](https://github.com/ef4/ember-code-snippet) appropriate `snippetRegexes` options that is set up in this addon's `ember-cli-build.js` can't handle angle brackets, which might be misleading: although the docs [do mention](https://ember-learn.github.io/ember-cli-addon-docs/docs/components/docs-demo#basic-usage) that:
> you can copy-paste this block to get going in your own app

but they don't mention that actually _any_ other way of invoking it, like naming `<DocsDemo>` yield something other than `demo`, using angle brackets or having example code in the same line as `<demo.example>` opening tag will provide with erroneous / unintended behavior.

I'll submit a PR to address angle bracket invocation but maybe one line in the [`<DocsDemo>` docs](https://ember-learn.github.io/ember-cli-addon-docs/docs/components/docs-demo) could mention that users should always use `demo` as the yielded hash's reference.

### PR DESCRIPTION
**Pull request by abel-n**
### Notes
Fixes #420
Closes #342 
Based on info in [nosloush's PR](https://github.com/ember-learn/ember-cli-addon-docs/pull/342), especially [dfreeman's suggestion](https://github.com/ember-learn/ember-cli-addon-docs/pull/342#issuecomment-517229597)

### Demo
image

### CODE REVIEW

File: test-apps/new-addon/tests/acceptance/snippets-test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    snippetIds.forEach(id => {
+      let snippet = `[data-test-id="${id}"]`;
+
+      assert.dom(`${snippet} pre`).includesText('<div id=\'foo\'>');
+      assert.dom(`${snippet} button`).hasAttribute('data-clipboard-text', /.*<div id='foo'>.*/);
+    });
+
+    let demo = '[data-test-id="your-angle-bracket-snippet-name.hbs"] ~ *:not([data-test-id])';
+    let secondDemo = `${demo}:last-child`;
```
Comment (abel-n) [inline test-apps/new-addon/tests/acceptance/snippets-test.js]: Let me note that I wanted to use the examples from the docs (or I'm just lazy) and the `<DocsDemo />` components don't provide with `data-test-id`s and I couldn't quickly find a better way to target them.

**Comment by scalvert** [comment]
I think this commit has introduced a test failure in master. Running `git bisect` on master flags this commit as the first time master's tests fail in CI.

The issue seems to be something to do with the `clipboardText` not being set correctly with the `snippetText` computed in the `docs-snippet` template. I haven't debugged further than this, but wanted to flag this here.

**Comment by abel-n** [comment]
I think I found the source of the issue, it's in [the commit before this merge commit](https://github.com/ember-learn/ember-cli-addon-docs/commit/5fb1cc8093ef530610345aa3938f2133112e91dd#diff-f7fa78c2855306dcc677ce08403e4a7bR84-R87) and the _Copy_ button actually doesn't work in `ember s`, there was just no test for it. The same button works as expected in the commit before that. I will open a PR to fix. :+1:

**Comment by abel-n** [comment]
[Opened a PR](https://github.com/ember-learn/ember-cli-addon-docs/pull/430) for that.

**Comment by scalvert** [comment]
Thanks man!

**Pull request merged**